### PR TITLE
Hotfix for #164: PHPCRLoader should be transactional

### PR DESCRIPTION
**Pull request by Ocramius**
Ping @electricmaxxx

See #164 
see doctrine/phpcr-odm#582

### CODE REVIEW

**Comment by stof** [comment]
any news ?

**Comment by lavoiesl** [comment]
@Ocramius Could you squash and rebase your commits?

File: tests/Doctrine/Tests/Common/DataFixtures/Executor/PHPCRExecutorTest.php [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+     * Ensures that the {@see \Doctrine\ODM\PHPCR\DocumentManager} class exists
+     */
+    private function loadDocumentManagerClass()
+    {
+
+        if (class_exists('Doctrine\ODM\PHPCR\DocumentManager')) {
+            return;
+        }
+
+        // hold my beer while I do some mocking
```
Comment (Ocramius) [inline tests/Doctrine/Tests/Common/DataFixtures/Executor/PHPCRExecutorTest.php]: I had to :-P

File: tests/Doctrine/Tests/Common/DataFixtures/Executor/PHPCRExecutorTest.php [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+        return $this->getMock('Doctrine\Common\DataFixtures\FixtureInterface', array(), array(), '', false);
+    }
+
+    /**
+     * Ensures that the {@see \Doctrine\ODM\PHPCR\DocumentManager} class exists
+     */
+    private function loadDocumentManagerClass()
+    {
+
+        if (class_exists('Doctrine\ODM\PHPCR\DocumentManager')) {
```
Comment (dbu) [inline tests/Doctrine/Tests/Common/DataFixtures/Executor/PHPCRExecutorTest.php]: should we not handle this with a require-dev?

Comment (Ocramius) [inline tests/Doctrine/Tests/Common/DataFixtures/Executor/PHPCRExecutorTest.php]: :-1: just a dependency for something I'd mock out anyway.

We're acting only on interfaces in any case

File: tests/Doctrine/Tests/Common/DataFixtures/Executor/PHPCRExecutorTest.php [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+    public function testExecuteSingleFixtureWithNoPurge()
+    {
+        $dm       = $this->getDocumentManager();
+        $executor = new PHPCRExecutor($dm);
+        $fixture  = $this->getMockFixture();
+
+        $fixture->expects($this->once())->method('load')->with($dm);
+        $dm
+            ->expects($this->once())
+            ->method('transactional')
```
Comment (dbu) [inline tests/Doctrine/Tests/Common/DataFixtures/Executor/PHPCRExecutorTest.php]: this is hiding https://github.com/symfony-cmf/symfony-cmf/issues/217 :-P

**Pull request closed without merging**